### PR TITLE
Docker compose

### PR DESCRIPTION
**Pull request by elainearbaugh**
Update cabot to use docker-compose and fix the tests

### CODE REVIEW

**Comment by elainearbaugh** [comment]
(WIP)

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,14 @@
 #!/usr/bin/env python
+import os
 from setuptools import setup, find_packages
 from os import environ as env
 
+requirements_file = os.path.join(os.path.dirname(__file__), 'requirements.txt')
+with open(requirements_file) as f:
+    requirements = [line.rstrip('\n')
+                    for line in f
+                    if line and not line.startswith('#')]
```
Comment (jimjh) [inline setup.py]: Have you considered using `pkg_resources.parse_requirements`?

e.g. https://github.com/Affirm/datatools/blob/master/dbimport/setup.py#L27

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -e
```
Comment (jimjh) [inline docker-entrypoint.sh]: I also recommend

```
set -o pipefail
set -o nounset
```

See [Coding Guidelines](https://github.com/Affirm/salt/blob/develop/CODING_GUIDELINES.md).

File: cabot/cabotapp/influx.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def get_data(pattern, selector='value',
              limit=settings.INFLUXDB_LIMIT,
              fetchall=False):
     '''
-    Query a metric and it's data from influxdb.
+    Query a metric and its data from influxdb.
```
Comment (jimjh) [inline cabot/cabotapp/influx.py]: 🖖

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+RUN mkdir /code
+
+WORKDIR /code
+
+RUN apt-get update && apt-get install -y \
+        python-dev \
+        libsasl2-dev \
+        libldap2-dev \
+        libpq-dev \
+        npm
```
Comment (jimjh) [inline Dockerfile]: I believe you need

```
DEBIAN_FRONTEND=noninteractive \
apt-get -qqy update && \
apt-get -qqy install python-dev ...
```

**Comment by elainearbaugh** [comment]
@Affirm/infrastructure please review?
- Copied docker-compose code from arachnys/cabot
- Fixed tests that were failing
- Added test for jenkins success since all the others were testing for failure

File: cabot/cabotapp/monitor.py [hunk 5]
```diff
@@ -28,14 +28,16 @@
...
-except:
-    DIMENSIONS = {}
+if CONNECTION:
+    try:
+        import boto.utils
+        _instance = boto.utils.get_instance_metadata(num_retries=2)
+        DIMENSIONS = {
+            'instance-id': _instance['instance-id'],
+        }
+        DIMENSIONS = {}
```
Comment (gregsterin) [inline cabot/cabotapp/monitor.py]: This wipes the dimensions we just set, no?

Comment (elainearbaugh) [inline cabot/cabotapp/monitor.py]: oops thanks for catching

File: setup.py [hunk 6]
```diff
@@ -1,6 +1,12 @@
...
 from setuptools import setup, find_packages
 from os import environ as env
+import pkg_resources
+
+requirements_file = os.path.join(os.path.dirname(__file__), 'requirements.txt')
+with open(requirements_file) as f:
+    requirements = pkg_resources.parse_requirements(f)
 
 # pull in active plugins
 plugins = env['CABOT_PLUGINS_ENABLED'].split(',') if 'CABOT_PLUGINS_ENABLED' in env else ["cabot_alert_hipchat", "cabot_alert_twilio", "cabot_alert_email"]
```
Comment (elainearbaugh) [inline setup.py]: Yeah not sure why they had CABOT_PLUGINS_ENABLED and requirements-pluginx.txt...
sorry which package?

Comment (elainearbaugh) [inline setup.py]: Guess it's not needed for the setup with docker-compose? I copied this part from arachnys

We don't use travis-ci anyways right? so maybe I should just get rid travis.yml...

Comment (srikiraju) [inline setup.py]: I don't really have any context about this, I just thought that change looked strange. I can offer a 🌮

Comment (elainearbaugh) [inline setup.py]: I got rid of it 🌯

File: Dockerfile (hunk 4 above)
Comment (elainearbaugh) [inline Dockerfile]: added env debian_frontend

**Comment by elainearbaugh** [comment]
merged with https://github.com/Affirm/cabot/pull/6

**Pull request closed without merging**